server_http.go

```diff
@@ -0,0 +1,220 @@
+package ekanite
+
+import (
+	"html/template"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/csrf"
+	"github.com/gorilla/securecookie"
+)
+
+// Server serves query client connections.
+type HttpServer struct {
+	iface    string
+	Searcher Searcher
+
+	addr     net.Addr
+	template *template.Template
+
+	Logger *log.Logger
+}
+
+// NewServer returns a new Server instance.
+func NewHttpServer(iface string, searcher Searcher) *HttpServer {
+	return &HttpServer{
+		iface:    iface,
+		Searcher: searcher,
+		Logger:   log.New(os.Stderr, "[httpserver] ", log.LstdFlags),
+	}
+}
+
+// Start instructs the Server to bind to the interface and accept connections.
+func (s *HttpServer) Start() error {
+
+	ln, err := net.Listen("tcp", s.iface)
+	if err != nil {
+		return err
+	}
+
+	s.addr = ln.Addr()
+
+	s.template, err = template.New("ServerTemplate").Parse(templateSource)
+	if err != nil {
+		ln.Close()
+		return err
+	}
+
+	CSRF := csrf.Protect(securecookie.GenerateRandomKey(32),
```

Comment:
I'm not quite sure what you are referring to - the Start func of HttpServer is called by cmd/ekanited in its initialization routine if the queryhttp flag is set. Protection  is indeed there to prevent [Cross Site Request Forgery](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_%28CSRF%29).